File: algorithms/0917.reverse-only-letters/solution.rb
# @param {String} s
# @return {String}
def reverse_only_letters(s)
  alph = [*('a'..'z'), *('A'..'Z')]
  letters = s.chars.select {|c| alph.include?(c) }
  s.chars.map {|c| alph.include?(c) ? letters.pop : c }.join
end

File: algorithms/0917.reverse-only-letters/solution.rs
impl Solution {
    pub fn reverse_only_letters(s: String) -> String {
        let mut letters: Vec<char> = s.chars().filter(|c| c.is_alphabetic()).collect();

        s.chars()
            .map(|c| {
                if c.is_alphabetic() { letters.pop().unwrap() } else { c }
            })
            .collect::<String>()
    }
}
